cfgov/paying_for_college/disclosures/scripts/process_cohorts.py

```diff
@@ -72,29 +65,43 @@ def rank_by_metric(school, cohort, metric):
 
 
 def run(single_school=None):
-    """Get percentile rankings for schools by control, degree, and state."""
+    """Get percentile rankings for schools by degree, control, and state."""
     count = 0
     starter = datetime.datetime.now()
-    base_query = build_cohorts()
+    base_query = build_base_cohorts()
     if single_school:
+        # a single_school query is allowed to sidestep exclusions
         base_query = School.objects.filter(pk=single_school)
     for school in base_query:
+        by_degree = {}
         by_state = {}
         by_control = {}
-        by_highest_degree = {}
         count += 1
         if count % 500 == 0:  # pragma: no cover
-            print(count)
-        state_cohort = STATE.get(school.state) if school.state else None
-        control_cohort = CONTROL.get(
-            school.control) if school.control else None
-        degree_cohort = DEGREES_HIGHEST.get(
-            school.degrees_highest) if school.degrees_highest else None
+            logger.info(count)
+        # degree_cohort is the default, national base cohort
+        degree_cohort = DEGREE_COHORTS.get(school.degrees_highest)
```

Comment:
```suggestion
        degree_cohort = DEGREE_COHORTS.get(school.degrees_highest, [])
```

The [`HIGHEST_DEGREES`](https://github.com/cfpb/cfgov-refresh/blob/6b3d002fa796473cc9d71ec1e2378421dbcdb025/cfgov/paying_for_college/models/disclosures.py#L39) dict only contains values 0-4, and so `DEGREE_COHORTS` gets initialized with only those 5 keys. But there are  many schools (for example 488785) with a different `degrees_highest`, which makes this line return `None`. Then down below when `degree_cohort` is used, it is unexpectedly null.

For example:

```
(cfgov-refresh) ~/Projects/cfgov-refresh $ PYTHONPATH=cfgov DJANGO_SETTINGS_MODULE=cfgov.settings.local python -c 'import django; django.setup(); from paying_for_college.disclosures.scripts.process_cohorts import run; run(488785)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/chosaka/Projects/cfgov-refresh/cfgov/paying_for_college/disclosures/scripts/process_cohorts.py", line 85, in run
    s for s in degree_cohort
TypeError: 'NoneType' object is not iterable
```

Reply:
Many thanks for this catch. 

We'd like to classify all schools with higher post-graduate programs as graduate-degree-granting, so I added a function to add such schools to our level '4' cohort:

```python
def get_grad_level(school):
    """Consider degrees higher than graduate level '4' as graduate degrees."""
    if int(school.degrees_highest) > 4:
        return '4'
    else:
        return school.degrees_highest
```